knn_demonstrations_test.ipynb

```diff
@@ -0,0 +1,323 @@
+{
```

Comment:
This is great, thanks! Can we perhaps move this file to a `tests/` directory?

It's currently good for testing but since it doesn't yet have a realistic dataset let's not have it at the top level (yet). We can still link to it from the README and, in the future, add it extensively in the docs + add a notebook that shows the benefits of kNN.